config/yaml_test.go

```diff
@@ -258,3 +259,33 @@ func TestGrumpyTextUnMarshallerParsing(t *testing.T) {
 		assert.EqualError(t, err, "Unknown character: DarkwingDuck")
 	})
 }
+
+func TestMergeUnmarshaller(t *testing.T) {
+	provider := NewYAMLProviderFromBytes(complexMapYaml, complexMapYamlV2)
+
+	ms := mapStruct{}
+	assert.NoError(t, provider.Get("mapStruct").PopulateStruct(&ms))
+
+	assert.NotNil(t, ms.MyMap)
+	assert.NotZero(t, len(ms.MyMap))
+
+	found := false
+	switch p := ms.MyMap["policy"].(type) {
+	case map[interface{}]interface{}:
```

Comment:
what is this case statement doing in the test? seems like a table test is a better option so that it's clear exactly what behavior you're asserting